openeo/rest/datacube.py

```diff
@@ -1480,8 +1480,10 @@ def download(self, outputfile: Union[str, pathlib.Path, None] = None, format: st
         :param options: Optional, file format options
         :return: None if the result is stored to disk, or a bytes object returned by the backend.
         """
-        if format == None:
-            format = guess_format(outputfile = outputfile)
+        if outputfile is None and format is None:
+            format = "GTiff"
+        else:
+            format = format or format = guess_format(outputfile)
```

Comment:
that's not what I meant syntax-wise, it's just
```python
format = format or guess_format(outputfile)
```

Reply:
but due to this `outputfile=None`  case I din't think of, it's probably easier to do the whole thing with:
```python
if not format:
    format = guess_format(outputfile) if outputfile else "GTiff"
```

Reply:
I have trouble understanding your code example.
Right now I changed it to
```python
        if outputfile is None and format is None:
            format = "GTiff"
        elif format is None:
            format = guess_format(outputfile)
```
to cover the behaviour you described above (default to GTiff as bytes when on outputfile is provided).

However only putting `format = format or guess_format(outputfile)` crashes tests in imagecollectionclient. should there be an `if` in front of it?

Reply:
> but due to this outputfile=None case I din't think of, it's probably easier to do the whole thing with:

Saw that only now, thanks! ignore the comment above then.